src/traversal/writeRelayUpdatePayload.js

```diff
@@ -187,6 +199,17 @@ function handleMerge(
     if (!Object.prototype.hasOwnProperty.call(payload, fieldName)) {
       continue;
     }
+
+    // if fieldName is part of deletedIDFieldName in some NODE_DELETE config
+    if (configs.some((config) => {
+      return config.type === RelayMutationType.NODE_DELETE &&
+        Array.isArray(config.deletedIDFieldName) &&
+        config.deletedIDFieldName.length > 0 &&
+        config.deletedIDFieldName[0] === fieldName;
+    })) {
```

Comment:
Style nit (non-blocking feedback): would probably make this:

```
if (configs.some(config => (
  config.type === ... &&
  Array.isArray(...) &&
  config ... &&
  config ...
)) {
  continue;
}
```

(ie. no parens around single arrow function param, and no need for explicit `return`)